aws/request/waiter_test.go

```diff
@@ -498,9 +514,10 @@ func TestWaiter_WithContext(t *testing.T) {
 	statuses := []int{http.StatusNotFound, http.StatusOK}
 
 	w := request.Waiter{
-		Name:        "TestWaiter",
-		MaxAttempts: 10,
-		Delay:       request.ConstantWaiterDelay(1 * time.Millisecond),
+		Name:             "TestWaiter",
+		MaxAttempts:      10,
+		Delay:            request.ConstantWaiterDelay(1 * time.Millisecond),
+		SleepWithContext: aws.SleepWithContext,
```

Comment:
I suggest making this the default option if the `SleepWithContext` is nil.  This would prevent needing to modify all of the generated waiters.